librz/core/cmd/cmd_debug.c

```diff
@@ -4411,24 +4407,64 @@ RZ_IPI RzCmdStatus rz_cmd_debug_add_bp_handler(RzCore *core, int argc, const cha
 
 // dbl
 RZ_IPI RzCmdStatus rz_cmd_debug_list_bp_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
-	RzOutputMode mode = state->mode;
-	switch (mode) {
-	case RZ_OUTPUT_MODE_STANDARD:
-		rz_bp_list(core->dbg->bp, 0);
-		break;
-	case RZ_OUTPUT_MODE_RIZIN:
-		rz_bp_list(core->dbg->bp, 1);
-		break;
-	case RZ_OUTPUT_MODE_JSON:
-		rz_bp_list(core->dbg->bp, 'j');
-		break;
-	case RZ_OUTPUT_MODE_QUIET:
-		rz_bp_list(core->dbg->bp, -1);
-		break;
-	default:
-		rz_warn_if_reached();
-		break;
+	rz_return_val_if_fail(state && core->dbg && core->dbg->bp, RZ_CMD_STATUS_ERROR);
+	RzBreakpointItem *b;
+	RzListIter *iter;
+	PJ *pj = state->d.pj;
+	RzTable *t = state->d.t;
+	rz_cmd_state_output_array_start(state);
+
+	rz_list_foreach (core->dbg->bp->bps, iter, b) {
+		switch (state->mode) {
+		case RZ_OUTPUT_MODE_STANDARD:
+			rz_cons_printf("0x%08" PFMT64x " - 0x%08" PFMT64x
+				       " %d %s %s %s %s %s cmd=\"%s\" cond=\"%s\" "
+				       "name=\"%s\" module=\"%s\"\n",
+				b->addr, b->addr + b->size, b->size,
+				rz_str_rwx_i(b->perm),
+				b->hw ? "hw" : "sw",
+				b->trace ? "trace" : "break",
+				b->enabled ? "enabled" : "disabled",
+				rz_bp_is_valid(core->dbg->bp, b) ? "valid" : "invalid",
+				rz_str_get(b->data),
+				rz_str_get(b->cond),
+				rz_str_get(b->name),
+				rz_str_get(b->module_name));
+			break;
+		case RZ_OUTPUT_MODE_TABLE:
+			rz_table_set_columnsf(state->d.t, "XXdsssssssss", "start addr", "end addr", "size", "perm", "hw/sw", "type", "state", "validity", "cmd", "cond", "name", "module");
+			t->showFancy = true;
```

Comment:
I don't think this is needed either. We should probably have some sort of global setting or something.

Reply:
Do we have such a global setting?

Reply:
TBF, the fancy output looks much more prettier than the default output.

Reply:
I do agree :D

Reply:
I have opened a PR for this #2027.